### DISCUSSION TITLE
Using an external sensor/fan via Arduino

### DISCUSSION THREAD

**Discussion by Jonny11bonk** [Rem0o/FanControl.Releases#2132] [Q&A]
Is there a way to monitor a temperature sensor hooked up to an Arduino board or any other? Let's say I would like to use an NTC sensor to measure my coolant temperature to not run the fans on idle. And yes, I don't have any pins on my motherboard for such sensors. Most of them don't have. The board could also be a Raspberry Pi Pico or an Arduino Leonardo (ATmega32u4) since both can communicate directly without any additional serial communication protocol (Like the AVR programmers).

**Comment by Rem0o** [comment]
I've seen some people develop their own small plugin with basic serial com to interface an arduino. Can't find an example right now,  but take a look at https://github.com/Rem0o/FanControl.Releases/wiki/Plugins.   I made coding a plugin as simple as possible. It is straight forward if you are familiar with programming in general.

Such a plugin would wrap a basic serial communication program, and expose it as a IPluginSensor.
